### PR TITLE
Change Memcached to verify ConfigMap exists instead of Secret.

### PR DESCRIPTION
**Pull request by devonboyer**
## What

Change `Memcached` to verify `ConfigMap` exists instead of `Secret`.

### To DO
- [ ] Merge https://github.com/Shopify/cloudbuddies/pull/446

### CODE REVIEW

**Comment by devonboyer** [comment]
```
[INFO][2017-10-13 15:28:06 -0400]	
[INFO][2017-10-13 15:28:06 -0400]	------------------------------------Phase 1: Initializing deploy------------------------------------
[INFO][2017-10-13 15:28:06 -0400]	All required parameters and files are present
[INFO][2017-10-13 15:28:06 -0400]	Context tierstaging found
[INFO][2017-10-13 15:28:07 -0400]	Namespace memcached-testing found
[INFO][2017-10-13 15:28:07 -0400]	Discovering templates:
[INFO][2017-10-13 15:28:07 -0400]	  - Memcached/memcached-my-app
[INFO][2017-10-13 15:28:07 -0400]	
[INFO][2017-10-13 15:28:07 -0400]	----------------------------Phase 2: Checking initial resource statuses-----------------------------
[INFO][2017-10-13 15:28:08 -0400]	Memcached/memcached-my-app                        not found
[INFO][2017-10-13 15:28:08 -0400]	
[INFO][2017-10-13 15:28:08 -0400]	------------------------------Phase 3: Predeploying priority resources------------------------------
[INFO][2017-10-13 15:28:08 -0400]	Deploying Memcached/memcached-my-app (timeout: 300s)
[INFO][2017-10-13 15:28:13 -0400]	Successfully deployed in 4.5s: Memcached/memcached-my-app
[INFO][2017-10-13 15:28:13 -0400]	
[INFO][2017-10-13 15:28:13 -0400]	
[INFO][2017-10-13 15:28:13 -0400]	----------------------------------Phase 4: Deploying all resources----------------------------------
[INFO][2017-10-13 15:28:13 -0400]	Deploying Memcached/memcached-my-app (timeout: 300s)
[INFO][2017-10-13 15:28:15 -0400]	Successfully deployed in 2.0s: Memcached/memcached-my-app
[INFO][2017-10-13 15:28:15 -0400]	
[INFO][2017-10-13 15:28:15 -0400]	------------------------------------------Result: SUCCESS-------------------------------------------
[INFO][2017-10-13 15:28:15 -0400]	Successfully deployed 1 resource
[INFO][2017-10-13 15:28:15 -0400]	
[INFO][2017-10-13 15:28:15 -0400]	Successful resources
[INFO][2017-10-13 15:28:15 -0400]	Memcached/memcached-my-app 
```

**Comment by devonboyer** [comment]
@stefanmb Do you have any insight into these build failures? They seem unrelated to my changes, and previous builds last week were passing while being tested against the same k8s version.

**Comment by stefanmb** [comment]
@devonboyer Yeah - @karanthukral reported these last week. They seem to stem from a `kubectl` version mismatch, I'm fixing it right now.

File: lib/kubernetes-deploy/kubernetes_resource/memcached.rb [hunk 1]
```diff
@@ -33,26 +33,22 @@ def exists?
     private
 
     def memcached_deployment_exists?
-      deployments, _err, st = kubectl.run("get", "deployments", "-l name=#{@name}", "-o=json")
+      deployment, _err, st = kubectl.run("get", "deployments", "memcached-#{@name}", "-o=json")
```
Comment (KnVerey) [inline lib/kubernetes-deploy/kubernetes_resource/memcached.rb]: I'm not very familiar with the memcachedbuddy code, but looking at the `ServiceName()` function it seems like there are a few ways the deployment name could be generated from the CR name. Am I wrong about that?

Comment (devonboyer) [reply lib/kubernetes-deploy/kubernetes_resource/memcached.rb]: You are right. This is written assuming https://github.com/Shopify/cloudbuddies/pull/446 has merged.

**Pull request merged**